review (#3)
jeel38/owasp-mastg: demos/android/MASVS-CODE/MASTG-DEMO-0054/MASTG-DEMO-0054.md

---
platform: android

> title: Testing Local Storage for Input Validation with Semgrep

Reviewer: ```suggestion
title: Use of Local Storage for Input Validation with semgrep
```

id: MASTG-DEMO-0054
code: [kotlin]
test: MASTG-TEST-0281
---

### Sample

> This code demonstrates improper use of local storage via `SharedPreferences.putString()` and `getString()` for storing sensitive or user-influenced data, including potentially dangerous input like HTML and JSON.

Reviewer: ```suggestion
The code snippet shows the improper use of local storage via `putString()` and `getString()` for storing sensitive data.
```

{{ MastgTest.kt # MastgTest_reversed.java }}

### Steps

Let's run @MASTG-TOOL-0110 rules against the sample code.

> {{ ../../../../rules/mastg-android-Local-Storage-for-Input-Validation.yml }}

Reviewer: ```suggestion
{{ ../../../../rules/mastg-android-local-storage-for-input-validation.yml }}
```

{{ run.sh }}

### Observation

The output file shows usages of string-based local storage in the code.

{{ output.txt }}

### Evaluation

> The test fails because `putString()` and `getString()` were used to store and retrieve structured or potentially user-controlled data, such as:

Reviewer: ```suggestion
The test fails because `putString()` and `getString()` were found in the code.
```
